devlib/utils/android.py

```diff
@@ -605,86 +590,32 @@ def run(self):
                 regexp = '({})'.format(regexp)
             logcat_cmd = '{} -e "{}"'.format(logcat_cmd, regexp)
 
-        logger.debug('logcat command ="{}"'.format(logcat_cmd))
-        self._logcat = self.target.background(logcat_cmd)
-
-        self._started.set()
+        logcat_cmd = get_adb_command(self.target.conn.device, logcat_cmd)
 
-        while not self._stopped.is_set():
-            line = self._logcat.stdout.readline(1024)
-            if line:
-                self._add_line(line)
+        logger.debug('logcat command ="{}"'.format(logcat_cmd))
+        self._logcat = pexpect.spawn(logcat_cmd, logfile=self._logfile)
 
     def stop(self):
```

Comment:
Should not this be called also from a dtor, i.e. the `__del__` method?

Reply:
I've never used that method before... but it could be useful since it means a broken user that forgets to call `stop` won't leave a `pexect` lying around. @setrofim what do you think? If we go for it though, I'll do it in a separate PR.

Reply:
Adding a call to `stop` inside `pexpect` is a good idea to make sure `logfile` is always closed properly, even if the client code neglects to explicitly close the monitor.

However, that would mean that `stop` could be called multiple times, so you should make sure that it would be safe to do so (that the second call does not raise errors).

And yeah, probably best to do this in a separate PR.